collect_performance.py

```diff
@@ -1,118 +1,205 @@
+
+    groupby_inner = groupby.copy()
+
+    match groupby_inner:
+        case None:
+            sdetails = get_submission_details(submission)
+            df["team"] = sdetails["team"]
+            df["model"] = sdetails["even_shorter_identifier"]
+            return df
+        case str():
+            if groupby_inner == "pooled":
+                groupby_inner = []
+            else:
+                groupby_inner = [groupby_inner]
+        case list():
+            pass
+        case _:
+            raise ValueError
+
+    df = get_target_data(submission / "eval", target = target)
+    if df.index.names != [None]:
+        df = df.reset_index()
+
+    if "year" in groupby_inner:
+        df["year"] = views_month_id_to_year(df["month_id"])
+    if "month" in groupby_inner:
+        df["month"] = views_month_id_to_month(df["month_id"])
+
+    if "month_id" not in groupby_inner:
+        df = df.drop(columns="month_id")
+    if unit not in groupby_inner:
+        df = df.drop(columns=unit)
+    if "window" not in groupby_inner:
+        df = df.drop(columns="window")
+
+    if across_submissions:
+        pass
+    else:
+        sdetails = get_submission_details(submission)
+        df["team"] = sdetails["team"]
+        df["model"] = sdetails["even_shorter_identifier"]
+        groupby_inner.extend(["team", "model"])
 
-def concat_eval(submission: str|os.PathLike, target: str, aggregation: str, metric: str) -> pd.DataFrame:
-    eval_data: list[str|os.PathLike] = list(submission.glob(f'**/{metric}_{aggregation}.parquet'))
-    eval_data: list[str|os.PathLike] = [f for f in eval_data if f.parts[-4] == target]
+    # Aggregate metric values
+    groupby_inner.append("metric")
+    df = df.set_index(groupby_inner)
+    df = df.groupby(level=groupby_inner, observed = True).mean()
+    groupby_inner.pop()
 
-    with open(submission/"submission_details.yml") as f:
-        submission_details = yaml.safe_load(f)
-    identifier = submission_details["even_shorter_identifier"]
-    team = submission_details["team"]
-    
-    if len(eval_data) == 0:
-        return print(f'No files to collect in submission {submission} for target {target} and aggregation {aggregation}.')
-    
-    dfs: list = []
-    for f in eval_data:
-        window = f.parts[-3]
-        target2 = f.parts[-4]
-        assert target == target2
+    # Pivot metrics to wide
+    df = df.pivot_table(values=["value"], index = groupby_inner, columns = "metric")
+    df.columns = df.columns.get_level_values(1).to_list()
 
-        sdf: pd.DataFrame = setup_eval_df(f, team, identifier, target, window)
-        
-        dfs.append(sdf)
-    
-    return pd.concat(dfs)
+    return df
 
-def merge_eval(submission: str|os.PathLike, target, aggregation) -> None:
-    metrics = ["crps", "ign", "mis"]
+def evaluation_table(submissions: str|os.PathLike, target: TargetType, groupby: str|list[str], save_to: str|os.PathLike = None, across_submissions: bool = False):
```

Comment:
It'll be nice to specify the return type of the function for uniformity across all the functions. 
def my_function(condition: bool) -> Union[pd.DataFrame, None, list]:
    if condition:
        return pd.DataFrame({'A': [1, 2, 3], 'B': [4, 5, 6]})
    elif not condition:
        return None
    else:
        return [1, 2, 3]